### PR TITLE
[integration] added script for multiVO certification

### PR DESCRIPTION
**Pull request by marianne013**
BEGINRELEASENOTES

*tests
NEW: Added a script intended for multi-VO certification that actually uses two different VOs. If approved would need a trello card as well.

ENDRELEASENOTES

### CODE REVIEW

File: tests/System/client_multivo.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+testdir=`echo $PWD`
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:3]: I might be missing something but:
```suggestion
testdir=$PWD
```

File: tests/System/client_multivo.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+]
+EOF
+
+
+EXEFILE="${testdir}/multiVOexe.sh"
+cat > ${EXEFILE} <<EOF
+echo $DIRACSITE
+ghostname=`hostname --long 2>&1`
+gipname=`hostname --ip-address 2>&1`
+echo $ghostname "has address" $gipname
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:24]: Does this do what you intend? Or did you mean:
```suggestion
echo \$DIRACSITE
ghostname=\$(hostname --long 2>&1)
gipname=\$(hostname --ip-address 2>&1)
echo \$ghostname "has address" \$gipname
```

File: tests/System/client_multivo.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+uname -a
+cat /etc/redhat-release
+EOF
+
+MYDATE=`date +%s`
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:29]: https://github.com/koalaman/shellcheck/wiki/SC2006
```suggestion
MYDATE=$(date +%s)
```

File: tests/System/client_multivo.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+fi
+
+echo -e "\nSubmitting job as a dteam user"
+dirac-wms-job-submit -f dteam_jobid.log multiVO.jdl
+sleep 30
+dirac-wms-job-status -f dteam_jobid.log
+
+echo -e "\nUploading file as a dteam_user"
+dirac-dms-add-file /dteam/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk
+if [[ "${?}" -ne 0 ]]; then
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:72]:
```suggestion
if dirac-dms-add-file "/dteam/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt" UKI-LT2-IC-HEP-disk; then
```

File: tests/System/client_multivo.sh [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+fi
+echo -e "\nRemoving file"
+dirac-dms-remove-files /gridpp/diraccert/testfile.${MYDATE}.txt
+
+
+export X509_USER_PROXY=${testdir}/dteam.proxy
+echo -e "\nChanging VO to dteam."
+echo "dirac-proxy-init -g dteam_user --out ${testdir}/dteam.proxy"
+dirac-proxy-init -g dteam_user --out ${testdir}/dteam.proxy
+if [[ "${?}" -ne 0 ]]; then
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:61]:
```suggestion
if dirac-proxy-init -g dteam_user --out "${testdir}/dteam.proxy"; then
```

File: tests/System/client_multivo.sh [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+fi
+
+echo -e "\nSubmitting job as a gridpp user"
+dirac-wms-job-submit -f gridpp_jobid.log multiVO.jdl
+sleep 30
+dirac-wms-job-status -f gridpp_jobid.log
+
+echo -e "\nUploading file as a gridpp_user"
+dirac-dms-add-file /gridpp/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk
+if [[ "${?}" -ne 0 ]]; then
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:50]:
```suggestion
if dirac-dms-add-file "/gridpp/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk"; then
```

File: tests/System/client_multivo.sh [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+env > testfile.${MYDATE}.txt
+
+
+echo -e "\nTesting first VO: gridpp"
+# just writing the proxy to a different file causes this command to fail
+# need to tell DIRAC beforehand where the user proxy will be
+export X509_USER_PROXY=${testdir}/gridpp.proxy
+echo "dirac-proxy-init -g gridpp_user --out ${testdir}/gridpp.proxy"
+dirac-proxy-init -g gridpp_user --out ${testdir}/gridpp.proxy
+if [[ "${?}" -ne 0 ]]; then
```
Comment (chrisburr) [inline tests/System/client_multivo.sh:39]:
```suggestion
if dirac-proxy-init -g gridpp_user --out "${testdir}/gridpp.proxy"; then
```

**Comment by marianne013** [comment]
Sorry, I had a previous incarnation which worked, then did an update to make it "better", but currently jobs in the certification instance don't seem to be going anywhere, so I didn't really test it (Friday afternoon and all). I see if I can rig some tests against my own instance and submit an update.

**Comment by marianne013** [comment]
I think I addressed all the comments, apart from the if statement. I think that's a style issue and the current implementation is easier to read than shellcheck's. I've tested it on my local server as much as possible, but as I don't have a dteam VO enabled, I can't test everything.

**Review by fstagni — APPROVED** [review]

File: tests/System/client_multivo.sh [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+echo "dirac-proxy-init -g gridpp_user --out ${testdir}/gridpp.proxy"
+dirac-proxy-init -g gridpp_user --out "${testdir}/gridpp.proxy"
+if [[ "${?}" -ne 0 ]]; then
+   echo "Could not acquire gridpp proxy. Giving up." 
+   exit 1
+fi
+
+echo -e "\nSubmitting job as a gridpp user"
+dirac-wms-job-submit -f gridpp_jobid.log multiVO.jdl
+sleep 30
```
Comment (andresailer) [inline tests/System/client_multivo.sh:46]: Add `echo "sleeping 30 s"` so people don't wonder why nothing happens?

**Pull request merged**